### DISCUSSION TITLE
Signal Archiving

### DISCUSSION THREAD

**Discussion by malins** [AsamK/signal-cli#1569] [Q&A]
Hello,

I'm looking for a solution to archive Signal conversations on a server, to comply with regulatory rules. Is it possible for signal-cli to act as a "linked device", which regularly pulls conversations from the Signal servers and stores it to some archive?

Thank you,
Manuel

**Comment by cuu508** [comment]
> Is it possible for signal-cli to act as a "linked device"

IIRC yes – signal-cli can act as a linked device, and you can link it to a "normal account" running on a phone. [When you initiate a linking procedure](https://github.com/AsamK/signal-cli/wiki/Linking-other-devices-(Provisioning)), signal-cli prints out a `tsdevice://` URL. I don't remember for sure, but I think I had to convert it to a QR code, and scan it from the app.

> which regularly pulls conversations from the Signal servers and stores it to some archive?

Yes, you can have signal-cli running continuously, and either 

* fish out incoming messages from signal-cli logs
* have another process listen for new messages over JSON-RPC or DBUS interface, and save them somewhere

Once you have a working setup, you will probably also want to monitor it in some way, to make sure that new messages are still being received and archived, and the system does not silently break.
